examples/distributed-solver/distributed-solver.cpp

```diff
@@ -250,35 +247,25 @@ int main(int argc, char* argv[])
     // case.
     Ainv->apply(b, x);
 
-    // Take timings.
-    comm.synchronize();
-    ValueType t_solver_apply_end = gko::experimental::mpi::get_walltime();
-
-    // Compute the residual, this is done in the same way as in the
-    // non-distributed case.
-    x_host->copy_from(x);
-    auto one = gko::initialize<vec>({1.0}, exec);
-    auto minus_one = gko::initialize<vec>({-1.0}, exec);
-    A_host->apply(minus_one, x_host, one, b_host);
-    auto res_norm = gko::initialize<vec>({0.0}, exec->get_master());
-    b_host->compute_norm2(res_norm);
-
     // Take timings.
     comm.synchronize();
     ValueType t_end = gko::experimental::mpi::get_walltime();
 
+    // Get the residual.
+    auto res_norm = gko::as<vec>(logger->get_residual_norm());
```

Comment:
Yes, but we use it only with `write`, so it doesn't matter. But thinking of it, maybe the convergence logger should store the result on the host (or copy it there on access). Do you have an opinion on that?

Reply:
it should keep the residual norm in the corresponding executor if it still uses the gko::Dense to keep. Otherwise, it is hard to notice those dense vectors are in different executor

Reply:
Which different executor?

Reply:
No, they should use the same executor as their solver counterparts. That is also how it worked before. I didn't change anything related to that.

Reply:
The above comments are my opinion on the question whether to store residual in the host or not.
They are not the comments to the code